### PR TITLE
Fixed #7 - Keep `WeakReference` to parent `ParseObject` in `ParseRelation`

### LINKED ISSUE #7: Keep `WeakReference` to parent `ParseObject` in `ParseRelation`
Will also need to keep reference to parent `objectId` and `className`, so we can recreate a pointer to parent if parent was GC'd.

### PR DESCRIPTION
**Pull request by sowrabh**
As suggested in the issue, keeping reference to parent objectId, className and recreating parent pointer when the parent is GC'ed.

### CODE REVIEW

File: Parse/src/main/java/com/parse/ParseRelation.java [hunk 1]
```diff
@@ -193,7 +208,13 @@ public void remove(T object) {
   }
 
   /* package for tests */ ParseObject getParent() {
-    return parent;
+    if(parent == null){
+      return null;
+    }
+    if(parent.get() == null){
+      return ParseDecoder.get().decodePointer(parentClassName, parentObjectId);
```
Comment (grantland) [inline Parse/src/main/java/com/parse/ParseRelation.java]: Let's just use `ParseObject.createWithoutData(String, String)` instead here so it'll be more concise.

**Comment by grantland** [comment]
This looks awesome! I just have a minor change that you could squash into your current commit and I can merge this in.

**Comment by sowrabh** [comment]
@grantland Thanks for the quick review. Addressed and squashed. Please have a look.

**Comment by grantland** [comment]
Thanks for your contribution!

**Comment by grantland** [comment]
@sowrabh If you're interested we'd love to send you some Parse swag as a thank you for your contribution! Just send me an email with your shipping address and t-shirt size, my email is on my GitHub profile page. Thanks again!

**Comment by sowrabh** [comment]
@grantland Thanks a bunch. Sent you the details on email.

**Pull request merged**